src/test/java/fi/aalto/cs/apluscourses/model/ExerciseTest.java

```diff
@@ -11,53 +16,101 @@
 
   @Test
   public void testExercise() {
-    Exercise exercise = new Exercise(987, "def");
+    Exercise exercise = new Exercise(987, "def", Collections.emptyList(), 0, 0, 0);
```

Comment:
an alternative could be having a constructor with fewer values and a bunch of setters to use, a least, this way it would be clear what value is being set in particular :grey_question:

Reply:
We want to keep (at least certain) fields `final` so adding setters to the model class itself is not an option.

A separate builder class (preferably with 'fluent interface') could be used, though (for other classes too). It has been on my mind but in that 'Refactor after Sprint 3' section.